### PR TITLE
feat: documentation for hub builder action

### PR DESCRIPTION
**Pull request by rutujasurve94**
Documentation for:

- `hub-builder` actions

Linked to Jira issue: Create Hub Builder action documentation [jira-270](https://jinaai.atlassian.net/browse/JINAHUB-270)

### CODE REVIEW

File: chapters/hub/hub-builder.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Jina Hub Builder
+
+Jina’s hub-builder is a simple interface for building & validating Jina Hub executors. It is built on top of jina hub interface. It can be used as a Github action in the CICD workflow, or via CLI.
```
Comment (alexcg1) [inline chapters/hub/hub-builder.md]: - executors -> Executors
- CICD -> CI/CD
- Github -> GitHub
- `It is built on top of jina hub interface` - is `jina hub` the command or the project here? If command, wrap in backticks. If project, capitalize (please see @FionnD's style guide)

File: chapters/hub/hub-builder.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Github action usage:
+One can use the `hub builder` as a part of the CI pipeline, to build and test images in the Pull Request. The following YAML could be directly used in the workflow `.github/workflows/hub-builder.yml`.
```
Comment (alexcg1) [inline chapters/hub/hub-builder.md]: - One -> You
- could -> can

File: chapters/hub/hub-builder.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+jobs:
+  hub-builder:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Jina Hub Image Builder
+        uses: jina-ai/hub-builder@master
+```
+On every new PR, the builder finds modifications in `manifest.yml` recursively and tries to build a Hub image from it, one by one. That means, when you update an image, you must change manifest.yml to trigger the build, e.g. you can simply bump version field in `manifest.yml`.
```
Comment (alexcg1) [inline chapters/hub/hub-builder.md]: - `tries to build a Hub image from it` - what's `it`? (we have two subjects - "file" and "modifications" so need to be more specific here)
- `one by one` - it's building hub images one by one? Or it goes through each modification one by one?
- Wrap manifest.yml in backticks
- e.g. -> i.e.
- the version field

File: chapters/hub/hub-builder.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+### Hub builder action accepts following parameters:
```
Comment (alexcg1) [inline chapters/hub/hub-builder.md]: - **the** following parameters:

File: chapters/hub/hub-builder.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+1. `push` (boolean): if push to Docker Hub and MongoDB
+2. `dockerhub_username` and `dockerhub_password`: user-name and password of docker registry
+3. `dockerhub_registry`: URL to the registry
+4. `slack_webhook`: webhook for Slack notification
+5. `jina_version`: version of Jina for building hub image
```
Comment (alexcg1) [inline chapters/hub/hub-builder.md]: **H**ub

File: chapters/hub/hub-builder.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Jina Hub Image Builder
+        uses: jina-ai/hub-builder@master
+```
+On every new PR, the builder finds modifications in `manifest.yml` recursively and tries to build a Hub image from it, one by one. That means, when you update an image, you must change manifest.yml to trigger the build, e.g. you can simply bump version field in `manifest.yml`.
+
+### Hub builder action accepts following parameters:
+1. `push` (boolean): if push to Docker Hub and MongoDB
+2. `dockerhub_username` and `dockerhub_password`: user-name and password of docker registry
```
Comment (alexcg1) [inline chapters/hub/hub-builder.md]: - user-name -> username
- docker -> Docker

File: chapters/hub/hub-builder.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+3. `dockerhub_registry`: URL to the registry
+4. `slack_webhook`: webhook for Slack notification
+5. `jina_version`: version of Jina for building hub image
+ 
+Note: the input argument jina_version is different from the version of jina for running the pod in the container. The later should be set in your`_jina_pod/requirements.txt`. The jina_version is the version of jina when building the pod image.
```
Comment (alexcg1) [inline chapters/hub/hub-builder.md]: - **P**od (all instances except for `_jina_pod/...`)
- version of **J**ina (both times)
- `The later` -> `The latter`
- Wrap `jina_version` in backticks

### Question

`the input argument jina_version is different from the version of jina for running the pod in the container` - is it **always** different? Or it **can** be different?

**Review by alexcg1 — CHANGES_REQUESTED** [review]

File: chapters/hub/hub-builder.md (hunk 7 above)
Comment (rutujasurve94) [reply chapters/hub/hub-builder.md]: It *can* be different, I've updated the documentation for clarity.
Thanks a lot for the review

File: chapters/hub/hub-builder.md [hunk 8]
```diff
@@ -18,16 +18,16 @@ jobs:
...
 1. `push` (boolean): if push to Docker Hub and MongoDB
-2. `dockerhub_username` and `dockerhub_password`: user-name and password of docker registry
+2. `dockerhub_username` and `dockerhub_password`: username and password of Docker registry
 3. `dockerhub_registry`: URL to the registry
 4. `slack_webhook`: webhook for Slack notification
-5. `jina_version`: version of Jina for building hub image
+5. `jina_version`: version of Jina for building Hub image
  
-Note: the input argument jina_version is different from the version of jina for running the pod in the container. The later should be set in your`_jina_pod/requirements.txt`. The jina_version is the version of jina when building the pod image.
+Note: the input argument `jina_version` can be different from the version of Jina for running the Pod in the container. The latter should be set in your `_jina_pod/requirements.txt`. The `jina_version` is the version of jina when building the Pod image.
```
Comment (alexcg1) [inline chapters/hub/hub-builder.md]: Just change:
`is the version of jina`

to
`is the version of Jina`

and then we're golden!

**Review by alexcg1 — CHANGES_REQUESTED** [review]
Just one teeny-tiny change :)

File: chapters/hub/hub-builder.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+On every new PR, the builder finds modifications in `manifest.yml` recursively and tries to build a Hub image using the latest files. That means, when you update an image, you must change `manifest.yml` to trigger the build, i.e. you can simply bump the version field in `manifest.yml`. This is done one by one for each modified Executor.
+
+### Hub builder action accepts the following parameters:
+1. `push` (boolean): if push to Docker Hub and MongoDB
+2. `dockerhub_username` and `dockerhub_password`: username and password of Docker registry
+3. `dockerhub_registry`: URL to the registry
+4. `slack_webhook`: webhook for Slack notification
+5. `jina_version`: version of Jina for building Hub image
+ 
+Note: the input argument `jina_version` can be different from the version of Jina for running the Pod in the container. The latter should be set in your `_jina_pod/requirements.txt`. The `jina_version` is the version of jina when building the Pod image.
```
Comment (Yongxuanzhang) [inline chapters/hub/hub-builder.md]:
```suggestion
Note: the input argument `jina_version` can be different from the version of Jina for running the Pod in the container. The latter should be set in your `_jina_pod/requirements.txt`. The `jina_version` is the version of Jina when building the Pod image.
```

File: chapters/hub/hub-builder.md (hunk 8 above)
Comment (Yongxuanzhang) [reply chapters/hub/hub-builder.md:30]: For minor issues, we could use 'Insert a suggestion' to save time. 😄 
image

**Review by Yongxuanzhang — COMMENTED** [review]
Great work. Learned a lot! 😄

File: chapters/hub/hub-builder.md (hunk 9 above)
Comment (rutujasurve94) [reply chapters/hub/hub-builder.md]: thanks a lot, I've incorporated the changes.

**Review by FionnD — APPROVED** [review]

**Review by alexcg1 — APPROVED** [review]

**Pull request merged**